src/arviz_plots/backend/matplotlib/__init__.py

```diff
@@ -200,6 +200,22 @@ def _filter_kwargs(kwargs, artist, artist_kws):
 
 
 # "geoms"
+def hist(
+    y, l_e, r_e, target, *, bottom=None, color=unset, facecolor=unset, edgecolor=unset, **artist_kws
+):
+    """Interface to matplotlib for a histogram bar plot."""
+    artist_kws.setdefault("zorder", 2)
+    bins = list(l_e) + [r_e[-1]]
+    midpoints = [(bins[i] + bins[i + 1]) / 2 for i in range(len(bins) - 1)]
+    if color is not unset:
+        if facecolor is unset:
+            facecolor = color
+        if edgecolor is unset:
+            edgecolor = color
+    kwargs = {"bottom": bottom, "color": facecolor, "edgecolor": edgecolor}
+    return target.bar(midpoints, y, **_filter_kwargs(kwargs, None, artist_kws))
```

Comment:
To minimize computation, I would do the following:

```
widths = np.asarray(r_e) - np.asarray(l_e)
...
return target.bar(l_e, y, width=widths, align="edge", _**filter_...)
```

The bin edges of a histogram don't necessarly need to be equispaced, so we do need to use array-like widths. We don't need to compute the midpoints though, using `align="edge"` we can use the left edges which we already have as x values.

Reply:
Okay, I'll add this change and push

Reply:
This one definitely works better, the gaps between histogram bars seem eliminated now 

![image](https://github.com/arviz-devs/arviz-plots/assets/24991131/5805a397-1b8e-45ea-8a90-a9bf63f39ac8)



![image](https://github.com/arviz-devs/arviz-plots/assets/24991131/4303c7da-dd71-45d1-986e-155746bc08e6)